seacatauth/otp/service.py

```diff
@@ -167,10 +179,24 @@ async def _delete_expired_totp_secrets(self):
 				"count": result.deleted_count
 			})
 
+	async def has_activated_totp(self, credentials_id: str) -> bool:
+		"""
+		Check if the user has TOTP activated from TOTPCollection. (For backward compatibility: check it from credentials (the old way))
+		"""
+		collection = self.StorageService.get(collection=self.TOTPCollection, obj_id=credentials_id)
+		secret = collection.get("__totp")
+
+		if secret is None:
+			credentials = await self.CredentialsService.get(credentials_id, include=frozenset(["__totp"]))
+			secret = credentials.get("__totp")
+		if secret is not None and len(secret) > 0:
+			return True
+		return False
+
 
-def authn_totp(dbcred, credentials):
+def authn_totp(credentials: dict, request_data: dict) -> bool:
 	try:
-		totp = pyotp.TOTP(dbcred['__totp'])
-		return totp.verify(credentials['totp'])
+		totp = pyotp.TOTP(credentials['__totp'])
+		return totp.verify(request_data['totp'])
 	except KeyError:
 		return False
```

Comment:
this authentication method should work with both these cases:
- TOTP saved in the credentials collection (as before)
- TOTP saved in the new TOTP collection